packages/integration/src/github/GitHubIntegration.ts

```diff
@@ -47,6 +47,13 @@ export class GitHubIntegration implements ScmIntegration {
   }
 
   resolveUrl(options: { url: string; base: string }): string {
-    return defaultScmResolveUrl(options);
+    // GitHub uses blob URLs for files and tree urls for directory listings. But
+    // there is a redirect from tree to blob for files, so we can always return
+    // tree urls here.
+    return defaultScmResolveUrl(options).replace('/blob/', '/tree/');
+  }
+
+  resolveEditUrl(url: string): string {
+    return url.replace('/blob/', '/edit/');
```

Comment:
What about `https://github.com/webmodules/blob/blob/master/README.md` tho? Tbh I think it's worth handling that edge-case xP, at least we don't need to support `blob/blob/blob` branches

Reply:
Hmm I guess that is an issue we should fix. Should be easy to fix with a simple regex, give me a second! Insert comic here that starts with this sentence but ends in existential dread.

Reply:
Done